CMakeLists.txt

```diff
@@ -2,36 +2,43 @@ cmake_minimum_required(VERSION 3.13.4)
 
 project("SVF")
 
-
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/.config.in ${CMAKE_CURRENT_SOURCE_DIR}/include/Util/config.h @ONLY)
+configure_file(${CMAKE_SOURCE_DIR}/.config.in ${CMAKE_BINARY_DIR}/include/Util/config.h)
 
 # To support both in- and out-of-source builds,
 # we check for the presence of the add_llvm_loadable_module command.
 # - if this command is not present, we are building out-of-source
 if(NOT COMMAND add_llvm_library)
-    if (DEFINED LLVM_DIR)
-        set(ENV{LLVM_DIR} "${LLVM_DIR}")
-    endif()
-    if (DEFINED ENV{LLVM_DIR})
         # We need to match the build environment for LLVM:
         # In particular, we need C++14 and the -fno-rtti flag
         set(CMAKE_CXX_STANDARD 14)
         set(CMAKE_CXX_STANDARD_REQUIRED ON)
         # add -std=gnu++14
         set(CMAKE_CXX_EXTENSIONS ON)
 
-        set(COMMON_FLAGS "-fPIC -Werror -Wall")
+        add_compile_options("-Werror" "-Wall")
 
-        if(CMAKE_BUILD_TYPE MATCHES "Debug")
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${COMMON_FLAGS} -O0 -fno-rtti")
```

Comment:
Have you considered debugging mode? I didn't find -O0 in your change?

Reply:
Yes I did.

`cmake` is already taking care of this and provides different modes: https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#default-and-custom-configurations
Each mode is associated with different build flags:
* Debug: `-g`
* MinSizeRel: `-Os -DNDEBUG`
* Release: `-O3 -DNDEBUG`
* RelWithDebInfo: `-O2 -g -DNDEBUG`

There is no need to explicitly specify them but only adapt them if required.

Reply:
@yuleisui  You mean the release build under Linux with the provided packages, i.e. the ones that are downloaded?
I can have a look into it.

Reply:
Yes, you can take a look (1) the cmakelist.txt you did in your PR and (2) the one on the current upstream.